commit 0f68018d06fe9cd574cad13a4731ea9f8fb58b55
Author: Nathan Scott <nathans@redhat.com>
Date:   Thu May 29 16:47:14 2014 +1000

    Dodge compiler-generated signed overflow bullet from Qwt
    
    Avoid a possible signed integer overflow in :
    /usr/include/QtCore/qvector.h: In member function ‘void QwtPlotHistogram::flushPolygon(QPainter*, double, QPolygonF&) const’:
    /usr/include/QtCore/qvector.h:330: warning: assuming signed overflow does not occur when assuming that (X - c) <= X is always true
    
    Looks like Qwt 6.1 has reworked this area significantly, and
    the error will also be resolved on our next upgrade - but in
    the interim, fix it locally.

diff --git a/src/libpcp_qwt/src/qwt_plot_histogram.cpp b/src/libpcp_qwt/src/qwt_plot_histogram.cpp
index 1fcf871..53d0551 100644
--- a/src/libpcp_qwt/src/qwt_plot_histogram.cpp
+++ b/src/libpcp_qwt/src/qwt_plot_histogram.cpp
@@ -532,7 +532,10 @@ void QwtPlotHistogram::flushPolygon( QPainter *painter,
             polygon += QPointF( baseLine, polygon.first().y() );
         }
         QwtPainter::drawPolygon( painter, polygon );
-        polygon.resize( polygon.size() - 2 );
+	int resize = polygon.size();
+	if ( resize > 1 )
+	    resize -= 2;
+        polygon.resize( resize );
     }
     if ( d_data->pen.style() != Qt::NoPen )
     {
